python/lsst/dax/ppdb/sql/_chunk_exporter.py

```diff
@@ -0,0 +1,187 @@
+        "GCS operations will not be performed. "
+        "Please install the 'google-cloud-storage' library to enable."
+    )
+
+
+class ChunkExporter(PpdbSql):
+    """Exports data from Cassandra to in-memory Parquet and uploads to GCS."""
+
+    def __init__(self, *args, **kwargs):
+        _LOG.info("Initializing ChunkExporter with args: %s, kwargs: %s", args, kwargs)
+        if storage:
+            if "GOOGLE_APPLICATION_CREDENTIALS" not in os.environ:
+                raise RuntimeError("Environment variable GOOGLE_APPLICATION_CREDENTIALS is not set.")
+            self.bucket_name = kwargs.pop("bucket")
+            self.folder_name = kwargs.pop("folder")
+            self.client = storage.Client()
+            self.bucket = self.client.bucket(self.bucket_name)
+        if "compression_format" in kwargs:
+            self.compression_format = kwargs.pop("compression_format")
+        else:
+            self.compression_format = "snappy"
+        super().__init__(*args, **kwargs)
+        self.schema_version = self._metadata.get(self.meta_schema_version_key)
+
+    def store(
+        self,
+        replica_chunk: ReplicaChunk,
+        objects: ApdbTableData,
+        sources: ApdbTableData,
+        forced_sources: ApdbTableData,
+        *,
+        update: bool = False,
+    ) -> None:
+        try:
+            gcs_paths: dict[str, dict[str, str]] = {}
+            for table_name, table_data in zip(
+                ["objects", "sources", "forced_sources"],
+                [objects, sources, forced_sources],
+            ):
+                _LOG.info("Processing %s", table_name)
+                if len(table_data.rows()) == 0:
+                    _LOG.info("Skipping %s: table is empty", table_name)
+                    continue
+
+                try:
+                    arrow_table = self._convert_to_arrow(table_data)
+                except Exception as e:
+                    _LOG.error("Failed to convert table to Arrow: %s", e)
+                    raise
+
+                _LOG.info(
+                    "Created Arrow Table with %d rows and %d columns",
+                    arrow_table.num_rows,
+                    arrow_table.num_columns,
+                )
+                memory_usage_mb = arrow_table.nbytes / 1_048_576
+                _LOG.info("Estimated memory usage: %.2f MB", memory_usage_mb)
+
+                if storage:
+                    gcs_path = self._make_gcs_path(table_name, replica_chunk.id)
+                    _LOG.info("Uploading %s to GCS path: %s", table_name, gcs_path)
+                    metadata = {
+                        "chunk_id": str(replica_chunk.id),
+                        "compression_format": self.compression_format,
+                        "exported_at": datetime.now(timezone.utc).isoformat() + "Z",
+                        "record_count": str(arrow_table.num_rows),
+                        "schema_version": str(self.schema_version),
```

Comment:
This should not be here. Data come from APDB and has APDB schema, not PPDB schema. 
Related issue is that even though you inherit from `PpdbSql` you do not actually use any regular tables, only `PpdbReplicaChunks`. I think you do not need any regular tables at all, so when PPDB database is created it should only make `PpdbReplicaChunks` and `metadata` tables. But you may also need to extend `PpdbReplicaChunks` with extra columns. So there are more methods that you need to override from the base class.

Reply:
> This should not be here. Data come from APDB and has APDB schema, not PPDB schema. Related issue is that even though you inherit from `PpdbSql` you do not actually use any regular tables, only `PpdbReplicaChunks`. I think you do not need any regular tables at all, so when PPDB database is created it should only make `PpdbReplicaChunks` and `metadata` tables. But you may also need to extend `PpdbReplicaChunks` with extra columns. So there are more methods that you need to override from the base class.

I think `PpdbSql` needs a fairly extensive amount of refactoring before I could inherit from it cleanly. I basically only need the parts that use the `PpdbReplicaChunks` table.

Reply:
> Data come from APDB and has APDB schema, not PPDB schema.

This is setting metadata on the object with the APDB schema version, isn't it?

Reply:
> Do you want me to try and do this?

Yes, do what you need to support your case.

> Isn't this the APDB schema version?

This is the version of the schema for which PPDB was created. APDB schema version is stored in APDB. For your case PPDB schema version is irrelevant, and it should not even have that version stored.

Reply:
> For your case PPDB schema version is irrelevant

The BigQuery ingest will need to know what version of the APDB was used to export the files. The cloud-based ingest scripts will absolutely need to look at this in order to load data correctly. So setting it on the object metadata seemed reasonable to me.

Reply:
To get APDB schema version you need to ask APDB. You are checking PPDB schema version, they are not the same and you PPDB should not even need a version.